keygen/src/xcr_comm.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _XCR_COMM_H
```

Comment:
I personally would combine this with `json_model.h` as `auth.{h,c}`.  But its fine to separate them if you prefer: `auth_model.{hcc}` and `auth_api.{h,c}`.